plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/impl/menu/AbstractMenu.java

```diff
@@ -81,9 +87,28 @@ public boolean isDisposed() {
 		}
 		return itemsText;
 	}
-	
+
 	public MenuItem getSWTWidget(){
 		return menuItem;
 	}
-	
+
+	/**
+	 * Returns menu items (child elements).
+	 * @return List of inner Menus.
+	 */
+	public List<Menu> getMenuItems() {
+		MenuItem[] items = ml.getItemsFromMenu(mh.getMenuFromMenuItem(menuItem));
+		List<Menu> menus = new ArrayList<>(items.length);
+		
+		for (MenuItem item : items){
+			String title = mh.getMenuItemText(item);
+			
+			if(title.isEmpty())				
+				continue;
+
+			List<String> path = mh.getMenuPath(item);
+			menus.add(new ShellMenu(path.toArray(new String[1])));
```

Comment:
Creating a ShellMenu inside AbstractMenu is incorrent. ShellMenu is a specific type of Menu, This method is available also for other types of Menus such as ContextMenu and ViewMenu. So if I call this method from ContextMenu I will get a list of ShellMenu instances.